app/views/payments/_form.html.erb

```diff
@@ -0,0 +1,33 @@
+<%= form_with(model: payment) do |form| %>
+  <% if payment.errors.any? %>
+    <div style="color: red">
+      <h2><%= pluralize(payment.errors.count, "error") %> prohibited this post from being saved:</h2>
+
+      <ul>
+        <% payment.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div>
+    <%= form.text_field :name, placeholder: "Name", required: true %>
+  </div>
+
+  <div>
+    <%= form.number_field :amount, placeholder: "Amount", required: true %>
+  </div>
+
+  <div class="checkbox-wrap">
+    <%= form.label :categories, style: "display: block" %>
+
+    <%= form.collection_check_boxes :category_ids, @categories, :id, :name %>
+    <br>
+     <%# <%= form.select :category_id, @categories.collect { |c| [c.name, c.id] } %>
+  </div>
+
+  <div>
+    <%= form.submit %>
+  </div>
```

Comment:
[REQUIRED] I still get an error when I click create payment while the input is empty. Kindly use form validation on Rails backend and the HTML form on the frontend.
![image](https://user-images.githubusercontent.com/62938735/235309631-648c8d55-64be-4aef-bdba-38dd0735dd82.png)

Reply:
I did the fix on the committed code, but did not update the live version viewed here. My apolgies, it is now rectified, along with an added validation for the categories :+1: